src/main/java/bio/terra/workspace/service/resource/controlled/ControlledResourceService.java

```diff
@@ -141,6 +141,15 @@ public String createAiNotebookInstance(
       ApiJobControl jobControl,
       String resultPath,
       AuthenticatedUserRequest userRequest) {
+    if (!resource.getAccessScope().equals(AccessScopeType.ACCESS_SCOPE_PRIVATE)) {
```

Comment:
nit: I'd move these checks inside `ControlledAiNotebookInstanceResource.validate()` so that it's always enforced at resource creation time, regardless of whether it goes through this service or not

Reply:
Hmmm, need to leave the privateIamRoles check here because it's not actually a part of the resource, it only exists during the creation flight.